pkg/api/v1/types.go

```diff
@@ -49,6 +49,9 @@ type Properties struct {
 	// AgentPoolProfiles (in): configuration of OpenShift cluster VMs.
 	AgentPoolProfiles []AgentPoolProfile `json:"agentPoolProfiles,omitempty"`
 
+	// AuthProfile (in): configures OpenShift authentication
+	AuthProfile AuthProfile `json:"authProfile,omitempty"`
```

Comment:
Can `IdentityProviders []IdentityProvider` be at the top level instead of instead `AuthProfile`?

Reply:
We will have more here, like Red Hat Insights, Metrics, Container Registry, etc. These are not `ItentitityProviders` so this is left to have a struct for all Authentication related configurables. But lets discuss on the call.

Reply:
I don't think I'm against getting rid of AuthProfile

Reply:
not against that we can keep it or not against that we should remove it? :)

Reply:
I like authProfile - if we think we will need to extend our auth configurables in the future we should stick with it.